lib/preload.js

```diff
@@ -127,8 +125,81 @@ const isEdisonZipArchive = (file) => {
   .catchReturn(false)
 }
 
+const createContainer = (docker, image, splashImage, dockerPort, proxy, edisonFolder) => {
+  const mounts = []
+  return docker.version()
+  .then((version) => {
+    if (splashImage) {
+      mounts.push(bindMount(splashImage, SPLASH_IMAGE_PATH_IN_DOCKER, version.ApiVersion))
+    }
+
+    const env = [
+      `HTTP_PROXY=${proxy || ''}`,
+      `HTTPS_PROXY=${proxy || ''}`,
+      `DOCKER_PORT=${dockerPort || ''}`
+    ]
+
+    if (edisonFolder) {
+      const partitions = getEdisonPartitions(edisonFolder)
+      env.push(`PARTITIONS=${JSON.stringify(partitions)}`)
+      PARTITION_NAMES.forEach((name) => {
+        const part = partitions[name]
+        mounts.push(bindMount(part.file, part.image, version.ApiVersion))
+      })
+    } else {
+      mounts.push(bindMount(image, DISK_IMAGE_PATH_IN_DOCKER, version.ApiVersion))
+    }
+    const containerOptions = {
+      Image: DOCKER_IMAGE_TAG,
+      Name: preload.CONTAINER_NAME,
+      AttachStdout: true,
+      AttachStderr: true,
+      OpenStdin: true,
+      Env: env,
+      HostConfig: {
+        Privileged: true
+      }
+    }
+    // Before api 1.25 bind mounts were going to into HostConfig.Binds
+    containerOptions.HostConfig[(compareVersions(version.ApiVersion, '1.25') >= 0) ? 'Mounts' : 'Binds'] = mounts
+    if (os.platform() === 'linux') {
+      containerOptions.HostConfig.NetworkMode = 'host'
+    } else {
+      containerOptions.HostConfig.NetworkMode = 'bridge'
+      containerOptions.ExposedPorts = {}
+      containerOptions.ExposedPorts[`${dockerPort}/tcp`] = {}
+      containerOptions.HostConfig.PortBindings = {}
+      containerOptions.HostConfig.PortBindings[`${dockerPort}/tcp`] = [{
+        HostPort: `${dockerPort}`,
+        HostIp: ''
+      }]
+    }
+    return docker.createContainer(containerOptions)
+  })
+}
+
+const isReadWriteAccessibleFile = (image) => {
+  return Promise.all([
+    fs.accessAsync(image, R_OK | W_OK),
+    fs.statAsync(image)
+  ])
+  .spread((_, stats) => {
```

Comment:
`Promise.join` is slightly better than `all+spread` when starting a promise chain.

Reply:
done